Apps/GatewayApi/src/MapUtils/CommentMapUtils.cs

```diff
@@ -0,0 +1,87 @@
+        /// </summary>
+        /// <param name="userComment">The UserComment to convert.</param>
+        /// <param name="cryptoDelegate">The Cryptographic delegate to use.</param>
+        /// <param name="key">The encryption key to use.</param>
+        /// <param name="autoMapper">The automapper to use.</param>
+        /// <returns>A Comment.</returns>
+        public static Comment ToDbModel(UserComment userComment, ICryptoDelegate cryptoDelegate, string key, IMapper autoMapper)
+        {
+            Comment comment = autoMapper.Map<UserComment, Comment>(userComment, opts =>
+                opts.AfterMap((src, dest) =>
+                    dest.Text = !string.IsNullOrEmpty(src.Text) ? cryptoDelegate.Encrypt(key, src.Text) : string.Empty));
+            return comment;
+        }
+
+        /// <summary>
+        /// Converts from Comment to UserComment.
+        /// </summary>
+        /// <param name="comment">The Comment to convert.</param>
+        /// <param name="cryptoDelegate">The Cryptographic delegate to use.</param>
+        /// <param name="key">The encryption key to use.</param>
+        /// <param name="autoMapper">The automapper to use.</param>
+        /// <returns>A UserComment.</returns>
+        public static UserComment CreateFromDbModel(Comment comment, ICryptoDelegate cryptoDelegate, string key, IMapper autoMapper)
+        {
+            UserComment userComment = autoMapper.Map<Comment, UserComment>(comment, opts =>
+                opts.AfterMap((src, dest) =>
+                    dest.Text = !string.IsNullOrEmpty(src.Text) ? cryptoDelegate.Decrypt(key, src.Text) : string.Empty));
+
+            return userComment;
+        }
+
+        /// <summary>
+        /// Converts from an enumerable of Comments to an enumerable of UserComments.
+        /// </summary>
+        /// <param name="comments">The enumerable of comments to convert.</param>
+        /// <param name="cryptoDelegate">The Cryptographic delegate to use.</param>
+        /// <param name="key">The encryption key to use.</param>
+        /// <param name="autoMapper">The automapper to use.</param>
+        /// <returns>An Enumerable of UserComments.</returns>
+        public static IEnumerable<UserComment> CreateListFromDbModels(IEnumerable<Comment> comments, ICryptoDelegate cryptoDelegate, string key, IMapper autoMapper)
+        {
+            IEnumerable<UserComment> userComments = autoMapper.Map<IEnumerable<Comment>, IEnumerable<UserComment>>(
+                comments, opts =>
+                    opts.AfterMap((src, dest) =>
+                    {
+                        foreach ((Comment comment, UserComment userComment) in src.Zip(dest))
+                        {
+                            userComment.Text = !string.IsNullOrEmpty(comment.Text) ? cryptoDelegate.Decrypt(key, comment.Text) : string.Empty;
+                        }
+                    }));
+
+            return userComments;
+        }
```

Comment:
I think we should do away with methods that convert enumerables of one type to enumerables of another type.

The same functionality can be achieved using a Select() query in LINQ, e.g.

```
dbComments.Payload.Select(c => CommentMapUtils.CreateFromDbModel(c, this.cryptoDelegate, key, this.autoMapper))
```

Reply:
That is Cleaner and Clearer :)